fieldpath/set.go

```diff
@@ -94,6 +94,20 @@ func (s *Set) Difference(s2 *Set) *Set {
 	}
 }
 
+// RecursiveDifference returns a Set containing elements which:
+// * appear in s
+// * do not appear in s2
+//
+// Compared to a regular difference, this recursively removes
+// all children from s
+// that are either children or members in s2
```

Comment:
Can we align this comment more with the one for `Difference`? I'm having a hard time understanding the difference (no pun intended). Feel free to adjust both comments, the existing one isn't amazingly clear.

Reply:
Can you check again?

The difference is, that a Difference only respects fields that are directly specified.
So to remove `a.b.c` and `a.b.d` from s, both need to be in s2.
But RecursiveDifference behaves more like deleting a folder that contains folders. It's enough to specify `a` in s2 and it's members are deleted recursively, or in the case of RecursiveDifference not added to the result.

I tried a few different explanations, but none was really good. I'm open for suggestions :-/